File: hackerRank/algorithms/AliceAndBobsSillyGame/alice-and-bobs-silly-game.py
# https://www.hackerrank.com/challenges/alice-and-bobs-silly-game/problem
def count_primes():
    r = 10**5+1     # 1 <= n <= 10**5
    sieve = [0] * r

    for i in range(2, r):
        if not sieve[i]:
            sieve[i] = 2
            for j in range(i*2, r, i):
                sieve[j] = 1
    return sieve


if __name__ == '__main__':
    primes = count_primes()
    n = int(input())
    for _ in range(n):
        n = int(input())
        print("Bob" if primes[:n+1].count(2) % 2 == 0 else "Alice")

File: hackerRank/algorithms/AliceAndBobsSillyGame/AliceAndBobsSillyGame.java
package algorithms.AliceAndBobsSillyGame;

import java.util.Scanner;

// https://www.hackerrank.com/challenges/alice-and-bobs-silly-game/problem
public class AliceAndBobsSillyGame {
    static String sillyGame(int n) {
        boolean isAliceMove = true;
        boolean[] sieve = new boolean[n+1];

        for (int i = 2; i <= n; i++) {
            if (!sieve[i]) {
                isAliceMove = !isAliceMove;
                for (int j = i; j <= n; j += i) {
                    sieve[j] = true;
                }
            }
        }
        return isAliceMove ? "Bob" : "Alice";
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        for(int i = 0; i < n; i++){
            System.out.println(sillyGame(in.nextInt()));
        }
    }
}
